### PR TITLE
Template fixes

### PR DESCRIPTION
**Pull request by DCamma**
About removing the variable for the local infra state: it is forced by the template and can never be different per environment, therefore imo is wrong to have it in a variable.

### CODE REVIEW

File: deploy/infrastructure/main.tf [hunk 1]
```diff
@@ -3,7 +3,7 @@ locals {
 
   {%- if app_url_type == "subdomain" %}
 
-  app_url = join(".", compact([var.app_subdomain, {% if dns_provider == "aws" %}data.aws_route53_zone.this.name{% elif dns_provider == "cloudflare" %}data.cloudflare_zone.this.name{% endif %}]))
+  app_url = join(".", compact([var.app_subdomain, var.zone_name]))
```
Comment (DCamma) [inline deploy/infrastructure/main.tf:6]: If I am not mistaken, in both cases we end up with the zone name, therefore I simplified

Comment (swibrow) [reply deploy/infrastructure/main.tf:6]: I think its fine to be a little more static in the way we set these base values

**Review by swibrow — COMMENTED** [review]
Made some comments:

I think the service_name key in the template should change since you'd think its related to a kubernetes service name which its not.

File: deploy/application/providers.tf [hunk 2]
```diff
@@ -61,7 +61,7 @@ data "terraform_remote_state" "infra_local" {
   backend = "s3"
   config = {
     bucket = var.tf_state_bucket
-    key    = var.infra_tf_state_key
+    key    = "@{{ infra_tf_state_key }}"
```
[the remark this answers is not in the document]
Comment (DCamma) [reply deploy/application/providers.tf:64]: why? there is no need for it to be dynamic, as I wrote in the PR description:

> About removing the variable for the local infra state: it is forced by the template and can never be different per environment, therefore imo is wrong to have it in a variable.

File: deploy/application/main.tf [hunk 3]
```diff
@@ -38,6 +38,7 @@ resource "helm_release" "app" {
       scaling_enabled        = var.scaling_enabled
       {%- if app_url_type == "subdomain" %}
       hostname               = data.terraform_remote_state.infra_local.outputs.app_url
+      service_name           = var.app_name
```
[the remark this answers is not in the document]
Comment (DCamma) [reply deploy/application/main.tf]: 👍🏼 changed to alb_group_name

**Review by swibrow — APPROVED** [review]
All good about the bucket ref

**Pull request merged**